src/backend/lang_c/c_simple_ast_builder.rs

```diff
@@ -0,0 +1,371 @@
+use std::collections::{HashMap, HashSet};
+use frontend::radeco_containers::RadecoFunction;
+use middle::ir::MOpcode;
+use middle::ssa::utils;
+use middle::ssa::ssastorage::{NodeData, SSAStorage};
+use middle::ssa::ssa_traits::{SSA, SSAExtra, SSAMod, SSAWalk, ValueInfo};
+use middle::ssa::cfg_traits::CFG;
+use super::c_simple_ast::{ValueNode, SimpleCAST, SimpleCASTEdge, ValueEdge, ActionEdge};
+use super::c_simple;
+use petgraph::visit::EdgeRef;
+use petgraph::graph::{Graph, NodeIndex, EdgeIndex, Edges, EdgeReference};
+use petgraph::{EdgeDirection, Direction, Directed};
+
+macro_rules! add_jump_to_cfg {
+    ($self: ident, $source: expr, $target: expr, $edge: expr) => {
+        let src_node = $self.action_map.get(&$source).unwrap();
```

Comment:
Please use expect wherever possible as it makes it easier to pinpoint the location of a panic.